pkg/controller/autoimport/apiserverurl.go

```diff
@@ -0,0 +1,103 @@
+package autoimport
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/stolostron/managedcluster-import-controller/pkg/constants"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func updateClusterURL(ctx context.Context, client client.Client,
+	managedCluster *clusterv1.ManagedCluster, autoImportSecret *corev1.Secret) error {
+	apiServerURL, err := getAPIServerURL(autoImportSecret)
+	if err != nil {
+		return err
+	}
+
+	if apiServerURL == "" {
+		return nil
+	}
+
+	apiServerURL = strings.TrimSuffix(apiServerURL, "/")
+
+	clusterCopy := managedCluster.DeepCopy()
+	for _, config := range clusterCopy.Spec.ManagedClusterClientConfigs {
+		if config.URL == apiServerURL {
+			return nil
+		}
+	}
+
+	clusterCopy.Spec.ManagedClusterClientConfigs = append(clusterCopy.Spec.ManagedClusterClientConfigs,
+		clusterv1.ClientConfig{
+			URL: apiServerURL,
+		})
+
+	klog.Infof("update the apiServerURL %v to the spec of managedCluster %v.", apiServerURL, clusterCopy.Name)
```

Comment:
The web UI only shows the first URL in the list, so I am not sure if append or prepend is proper here, do we want the UI always show the URL in the bootstrap kubeconfig?

Reply:
not required to show which URL.  I think it is ok to always show the first URL.